model/atmosphere/dycore/tests/dycore_stencil_tests/test_compute_horizontal_advection_of_rho_and_theta.py

```diff
@@ -185,42 +184,36 @@ def input_data(self, grid):
         if np.any(grid.connectivities[dims.E2CDim] == -1):
             pytest.xfail("Stencil does not support missing neighbors.")
 
-        p_vn = random_field(grid, dims.EdgeDim, dims.KDim, dtype=wpfloat)
-        p_vt = random_field(grid, dims.EdgeDim, dims.KDim, dtype=vpfloat)
-        pos_on_tplane_e_1 = random_field(grid, dims.EdgeDim, dims.E2CDim, dtype=wpfloat)
-        pos_on_tplane_e_1_new = as_1D_sparse_field(pos_on_tplane_e_1, dims.ECDim)
-        pos_on_tplane_e_2 = random_field(grid, dims.EdgeDim, dims.E2CDim, dtype=wpfloat)
-        pos_on_tplane_e_2_new = as_1D_sparse_field(pos_on_tplane_e_2, dims.ECDim)
-        primal_normal_cell_1 = random_field(grid, dims.EdgeDim, dims.E2CDim, dtype=wpfloat)
-        primal_normal_cell_1_new = as_1D_sparse_field(primal_normal_cell_1, dims.ECDim)
-        dual_normal_cell_1 = random_field(grid, dims.EdgeDim, dims.E2CDim, dtype=wpfloat)
-        dual_normal_cell_1_new = as_1D_sparse_field(dual_normal_cell_1, dims.ECDim)
-        primal_normal_cell_2 = random_field(grid, dims.EdgeDim, dims.E2CDim, dtype=wpfloat)
-        primal_normal_cell_2_new = as_1D_sparse_field(primal_normal_cell_2, dims.ECDim)
-        dual_normal_cell_2 = random_field(grid, dims.EdgeDim, dims.E2CDim, dtype=wpfloat)
-        dual_normal_cell_2_new = as_1D_sparse_field(dual_normal_cell_2, dims.ECDim)
-        p_dthalf = wpfloat("2.0")
-
-        rho_ref_me = random_field(grid, dims.EdgeDim, dims.KDim, dtype=vpfloat)
-        theta_ref_me = random_field(grid, dims.EdgeDim, dims.KDim, dtype=vpfloat)
-        z_grad_rth_1 = random_field(grid, dims.CellDim, dims.KDim, dtype=vpfloat)
-        z_grad_rth_2 = random_field(grid, dims.CellDim, dims.KDim, dtype=vpfloat)
-        z_grad_rth_3 = random_field(grid, dims.CellDim, dims.KDim, dtype=vpfloat)
-        z_grad_rth_4 = random_field(grid, dims.CellDim, dims.KDim, dtype=vpfloat)
-        z_rth_pr_1 = random_field(grid, dims.CellDim, dims.KDim, dtype=vpfloat)
-        z_rth_pr_2 = random_field(grid, dims.CellDim, dims.KDim, dtype=vpfloat)
-        z_rho_e = random_field(grid, dims.EdgeDim, dims.KDim, dtype=wpfloat)
-        z_theta_v_e = random_field(grid, dims.EdgeDim, dims.KDim, dtype=wpfloat)
+        p_vn = data_alloc.random_field(grid, dims.EdgeDim, dims.KDim, dtype=ta.wpfloat)
+        p_vt = data_alloc.random_field(grid, dims.EdgeDim, dims.KDim, dtype=ta.vpfloat)
+        pos_on_tplane_e_1 = data_alloc.random_field(grid, dims.ECDim, dtype=ta.wpfloat)
+        pos_on_tplane_e_2 = data_alloc.random_field(grid, dims.ECDim, dtype=ta.wpfloat)
+        primal_normal_cell_1 = data_alloc.random_field(grid, dims.ECDim, dtype=ta.wpfloat)
+        dual_normal_cell_1 = data_alloc.random_field(grid, dims.ECDim, dtype=ta.wpfloat)
+        primal_normal_cell_2 = data_alloc.random_field(grid, dims.ECDim, dtype=ta.wpfloat)
+        dual_normal_cell_2 = data_alloc.random_field(grid, dims.ECDim, dtype=ta.wpfloat)
+        p_dthalf = ta.wpfloat("2.0")
```

Comment:
do we need type cast here? I thought that simply putting `p_dthalf = 2.0` would have been enough, but I could be wrong